src/java/org/apache/cassandra/db/Keyspace.java

```diff
@@ -743,22 +767,30 @@ public AbstractReplicationStrategy getReplicationStrategy()
 
     public static Iterable<Keyspace> all()
     {
-        return Iterables.transform(Schema.instance.getKeyspaces(), keyspaceTransformer);
+        return Iterables.transform(Schema.instance.getKeyspaces(), Keyspace::open);
+    }
+
+    /**
+     * @return a {@link Stream} of all keyspaces existing/open {@link Keyspace} instances
+     */
+    public static Stream<Keyspace> allExisting()
```

Comment:
should this be `Iterable` just to match all methods?

Reply:
the stream would at least avoid extra allocation from collecting, and also means we only call getKeyspaceInstance when we need to.

Reply:
for me: stated in JIRA; leaving here to isolate the change.  Over time more things maybe should switch over and all() might go away.